src/dcap_provider.cpp

```diff
@@ -89,6 +94,28 @@ static std::string get_env_variable(std::string env_variable)
     }
 }
 
+static std::string get_collateral_version()
+{
+    std::string collateral_version = get_env_variable(ENV_AZDCAP_COLLATERAL_VER);
+    if (!collateral_version.empty())
+    {
+        if (!collateral_version.compare("v1") && !collateral_version.compare("v2"))

```

Comment:
That will break comparability with the existing service so I'd like to leave it as-is. I know the service side changes are on their way but just in case wouldn't we rather fail due to supplying down-rev collateral rather than with 404's?